### DISCUSSION TITLE
SSL routines::ca key too small error wirh openSSL3

### DISCUSSION THREAD

**Discussion by reguripradeep** [openssl/openssl#24153] [Q&A]
I am getting "SSL routines::ca key too small" error when access https. The same CA certificate (with a key of size 1024) was working fine with OpenSSL 1.0.2zj but is not working when i upgraded to 3.2.0. I tried setting -DOPENSSL_TLS_SECURITY_LEVEL=1 while configuring openSSL compilation but it id not work.  do we require a CA cert with a key size of min 2048 with OpenSSL3 or is there any other way to get the same behavior as with OpenSSL1.0.2x .

openssl version -a
OpenSSL 3.3.0 9 Apr 2024 (Library: OpenSSL 3.3.0 9 Apr 2024)
built on: Mon Apr 15 15:13:50 2024 UTC
platform: VC-WIN64A
options:  bn(64,64)
compiler: cl  /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MD /W3 /wd4090 /nologo /O2 -DL_ENDIAN -DOPENSSL_PIC -D"OPENSSL_BUILDING_OPENSSL" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"NDEBUG" -DOPENSSL_TLS_SECURITY_LEVEL=1

**Comment by t8m** [comment]
Have you tried  `-DOPENSSL_TLS_SECURITY_LEVEL=0` ?

**Reply by reguripradeep** [reply]
No, i tried with LEVEL 1 only. LEVEL=0 is allowing everything right? i want a minimum security so i tried with 1

**Reply by reguripradeep** [reply]
it is working fine with LEVEL=0.

**Comment by reguripradeep** [comment]
it is working fine with LEVEL=0. So i recompiled again with LEVEL=1 and it is working too. probably i might not have copied the recompiled libs properly. Hence closing the discussion
